packages/contracts-bedrock/test/Authorization.t.sol

```diff
@@ -0,0 +1,266 @@
+        _addSpec("SuperchainConfig", _getSel("paused()"), false);
+        _addSpec("SuperchainConfig", _getSel("unpause()"), false);
+        _addSpec("SuperchainConfig", _getSel("version()"), false);
+
+        // SystemConfig
+        _addSpec("SystemConfig", _getSel("UNSAFE_BLOCK_SIGNER_SLOT()"), false);
+        _addSpec("SystemConfig", _getSel("VERSION()"), false);
+        _addSpec("SystemConfig", _getSel("batcherHash()"), false);
+        _addSpec("SystemConfig", _getSel("gasLimit()"), false);
+        _addSpec("SystemConfig", SystemConfig.initialize.selector, false);
+        _addSpec("SystemConfig", SystemConfig.minimumGasLimit.selector, false);
+        _addSpec("SystemConfig", _getSel("overhead()"), false);
+        _addSpec("SystemConfig", _getSel("owner()"), false);
+        _addSpec("SystemConfig", _getSel("renounceOwnership()"), true);
+        _addSpec("SystemConfig", SystemConfig.resourceConfig.selector, false);
+        _addSpec("SystemConfig", _getSel("scalar()"), false);
+        _addSpec("SystemConfig", SystemConfig.setBatcherHash.selector, false);
+        _addSpec("SystemConfig", SystemConfig.setGasConfig.selector, false);
+        _addSpec("SystemConfig", SystemConfig.setGasLimit.selector, false);
+        _addSpec("SystemConfig", SystemConfig.setResourceConfig.selector, false);
+        _addSpec("SystemConfig", SystemConfig.setUnsafeBlockSigner.selector, false);
+        _addSpec("SystemConfig", _getSel("transferOwnership(address)"), false);
+        _addSpec("SystemConfig", SystemConfig.unsafeBlockSigner.selector, false);
+        _addSpec("SystemConfig", _getSel("version()"), false);
+    }
+
+    function _getSel(string memory _name) internal returns (bytes4) {
+        return bytes4(keccak256(abi.encodePacked(_name)));
+    }
+
+    function _addSpec(string memory _name, bytes4 _sel, bool _auth) internal {
+        specs[_name][_sel] = Spec({ name: _name, sel: _sel, auth: _auth });
+        numEntries[_name]++;
+    }
+
+    function testContractAuth() public {
+        Abi[] memory abis = _getL1ContractFunctionAbis();
+
+        for (uint256 i = 0; i < abis.length; i++) {
+            string memory contractName = abis[i].contractName;
+            assertEq(
+                abis[i].entries.length, numEntries[contractName], "Authorization_Test: invalid number of ABI entries"
+            );
+
+            for (uint256 j = 0; j < abis[i].entries.length; j++) {
+                AbiEntry memory abiEntry = abis[i].entries[j];
+                console.log(
+                    "Checking auth spec for %s: %s(%x)", contractName, abiEntry.fnName, uint256(uint32(abiEntry.sel))
+                );
+                Spec memory spec = specs[contractName][abiEntry.sel];
+                assertTrue(spec.sel != bytes4(0), "Authorization_Test: missing spec definition");
+                assertEq(abiEntry.sel, spec.sel, "Authorization_Test: invalid ABI");
+            }
```

Comment:
Maybe I'm missing something, but shouldn't there be an actual pranked call to each of these functions to validate the auth?

Reply:
OK, I talked with @mds1 the other day about this, and now understand the value as being that it ensures that a spec exists for all methods.

In the future we can consider extending this to actually test the spec, but for now this seems good.

Reply:
We already have forge tests that do this for every contract.
The aim of these tests is to make sure that there's an auth specification for every interaction with the L1 contracts. It's purely meant for documentation.